Assets/_Code/MenuKeybindings.cs

```diff
@@ -58,24 +68,37 @@ void LoadList()
             TextPressAKey.SetActive(true);
             if (Input.anyKeyDown)                                                            //If a key has been pressed
             {
-                foreach(KeyCode kc in Enum.GetValues( typeof(KeyCode)))                     //ForEach posible key
+                var pressedKey = InputManager.GetPressedKey();
+                // Check if that key is already used.
+                string alreadyBound = inputManager.GetButtonForKey(pressedKey);
+                if (string.IsNullOrEmpty(alreadyBound))
                 {
-                    if(Input.GetKeyDown(kc))                                                //Check if this key is pressed down
+                    // The key is still free so assign it and reload the ui.
+                    inputManager.SetButtonForKey(buttonToRebind, pressedKey);
+                    LoadList();
+                }
+                else if (alreadyBound != buttonToRebind)
+                {
+                    // Create a modal dialog with a text and two buttons.
+                    var text = string.Format(MsgAlreadyAssigned, pressedKey.ToString(), alreadyBound, buttonToRebind);
+                    // The yes button will perform the binding and reload the ui.
+                    var buttonName = buttonToRebind;
+                    var yes = new ModalDialogButton
                     {
-                        if(inputManager.SetButtonForKey(buttonToRebind, kc))                //Set the key, and check if this key has been used before
-                        {
-                            buttonToLabel[buttonToRebind].color = new Color(1f, 0.5f, 0.16f);  //Warn user that this key has more than one function
-                        }
-                        else
+                        Label = "Yes", Action = () =>
                         {
-                            buttonToLabel[buttonToRebind].color = Color.black;              //Key hasn't been used for something, so just make the color default black
+                            inputManager.SetButtonForKey(buttonName, pressedKey);
+                            LoadList();
                         }
-                        buttonToLabel[buttonToRebind].text = System.Convert.ToString(kc);                 //Set the 
-                        buttonToRebind = null;
-                        TextPressAKey.SetActive(false);
-                        break;
-                    }
+                    };
+                    // The no button will do nothing.
+                    var no = new ModalDialogButton { Label = "No" };
+
+                    // Show the created dialog.
+                    ModalDialog.Instance().ShowDialog(text, new[] {yes, no});
```

Comment:
Maybe also an option 'Ignore' so you would be able to do double keybindings?

Reply:
And what would be the purpose of duplicate keybindings?